### PR TITLE
Add parameter decoding CLI tool to F` GDS

### PR DESCRIPTION
**Pull request by Brian-Campuzano**
| | |
|:---|:---|
|**_Related Issue(s)_**| nasa/fprime#4750 |
|**_Has Unit Tests (y/n)_**| Yes |
|**_Documentation Included (y/n)_**| Yes (CLI help, docstrings) |
|**_Generative AI was used in this contribution (y/n)_**| Yes |

---
## Change Description

Adds `fprime-prm-decode` CLI tool to decode binary parameter database (`.dat`) files into human-readable formats (JSON, text, CSV). This provides the inverse operation of `fprime-prm-write`.

**Changes:**
- New `fprime-prm-decode` command-line tool
- Supports three output formats:
  - JSON: Round-trip compatible with `fprime-prm-write` input format
  - Text: Human-readable format with component grouping, types, and IDs
  - CSV: Spreadsheet-compatible format with columns: Component, Parameter, Value, Type, ID
- Comprehensive error handling for corrupt files, missing parameters, and deserialization failures
- Format conversion logic to ensure decoded JSON matches encoder expectations for round-trip compatibility

## Rationale

`fprime-prm-write` can encode parameter JSON to binary .dat files, but there's no way to decode them back. This makes it difficult to:

- Inspect deployed parameter databases (only raw hex dumps available)
- Verify encoded .dat files contain expected values
- Debug parameter-related issues in production systems
- Extract parameters from existing .dat files for modification

This tool completes the parameter workflow by enabling developers to inspect, verify, and debug parameter databases.

## Testing/Review Recommendations

**Unit Tests (14 tests, all passing):**
- Round-trip encode/decode verification
- All output formats (JSON, text, CSV)
- Error handling (invalid delimiter, unknown param ID, incomplete data)
- Complex type conversion (arrays, structs, nested structures)
- Empty file handling

**Review Recommendation:**
- `src/fprime_gds/common/tools/params.py`: Review decode functions and `main_decode()` entry point
- `pyproject.toml`: Review `fprime-prm-decode` CLI command registration
- `test/fprime_gds/common/tools/test_prm_decode.py`: Review 14 new unit tests

## Future Work

None planned.

## AI Usage (see [policy](https://github.com/nasa/fprime/blob/devel/AI_POLICY.md))
Claude (Sonnet 4.5) was used extensively in this contribution for:

* Initial implementation: Generated decoder functions, format converters, and CLI interface
* Test development: Created comprehensive unit test suite covering edge cases

All AI-generated code was reviewed, tested, and verified by myself. Round trip conversion of a parameter database was done manually using a flight like database.

### CODE REVIEW

File: src/fprime_gds/common/tools/params.py [hunk 1]
```diff
@@ -246,5 +251,298 @@ def convert_json(json_file: Path, dictionary: Path, output: Path, output_format:
...
+    while offset < len(dat_bytes):
+        # Check for delimiter
+        if dat_bytes[offset] != 0xA5:
+            raise RuntimeError(
+                f"Invalid delimiter at offset {offset}: expected 0xA5, got {dat_bytes[offset]:#x}"
+            )
+        offset += 1
+
+        # Read record size (4 bytes, big endian)
+        if offset + 4 > len(dat_bytes):
```
Comment (thomas-bc) [inline src/fprime_gds/common/tools/params.py:279]: I'm seeing hardcoded widths and such. You should instead be using the loaded dictionary types. See my other comment on how to do that. Once you do have it, you should be able to do something like

```python
from fprime_gds.common.utils.config_manager import ConfigManager

record_id_size = ConfigManager().get_type("FwDpIdType").getSize()
```

File: src/fprime_gds/common/tools/params.py [hunk 2]
```diff
@@ -246,5 +251,298 @@ def convert_json(json_file: Path, dictionary: Path, output: Path, output_format:
...
+    This is the inverse operation of fprime-prm-write.
+    """
+    arg_parser = ArgumentParser()
+
+    arg_parser.add_argument(
+        "dat_file", type=Path, help="The .dat file to decode", default=None
+    )
+    arg_parser.add_argument(
+        "--dictionary",
+        "-d",
```
Comment (thomas-bc) [inline src/fprime_gds/common/tools/params.py:485]: With that argument, you can leverage the pre-existing `Dictionaries` class which can load all the dictionary types into a global config singleton, which will help processing without hardcoding type widths and such.

See https://github.com/nasa/fprime-gds/blob/620cc301079267cb6bc881761140173dfc37a7f7/src/fprime_gds/executables/data_products.py#L26-L28

**Review by thomas-bc — COMMENTED** [review]
Leveraging the global config singleton would help a lot with making these more portable. I wonder if you could prompt your AI to do that - it should be able to handle it fairly easily since other places in the repo are doing the same thing quite a lot. 

As long as UTs pass, we should be fine.

But I'm also happy to take this in as is since the `encode()` variant doesn't currently do it.

**Review by thomas-bc — APPROVED** [review]
Looks good - let's merge this as is and I will capture a follow-up ticket to consolidate both scripts with Dictionary loading etc.

**Comment by thomas-bc** [comment]
Follow up ticket: https://github.com/nasa/fprime/issues/4792
